src/Microsoft.VisualStudio.Threading.Tests/AsyncReaderWriterLockTests.cs

```diff
@@ -1813,6 +1813,49 @@ public void WriteLockAsyncSynchronousReleaseAllowsOtherWriters()
             Assert.True(firstLockTask.Wait(TestTimeout)); // rethrow any exceptions
         }
 
+        [StaFact]
+        public void WriteLockDisposingShouldNotBlockByOtherWriters()
+        {
+            var firstLockToRelease = new AsyncManualResetEvent();
+            var firstLockAccquired = new AsyncManualResetEvent();
+            var firstLockToDispose = new AsyncManualResetEvent();
+            var firstLockTask = Task.Run(async delegate
+            {
+                using (var firstLock = await this.asyncLock.WriteLockAsync())
+                {
+                    firstLockAccquired.Set();
+                    await firstLockToRelease.WaitAsync();
+                    await firstLock.ReleaseAsync();
+                    await firstLockToDispose.WaitAsync();
+                }
+            });
+
+            var secondLockReleased = new AsyncManualResetEvent();
+
+            var secondLockTask = Task.Run(async delegate
+            {
+                await firstLockAccquired.WaitAsync();
+                var awaiter = this.asyncLock.WriteLockAsync().GetAwaiter();
+                Assert.False(awaiter.IsCompleted);
+                awaiter.OnCompleted(() =>
+                {
+                    firstLockToDispose.Set();
+
+                    // We must block the thread synchorizely, so it won't release the NonConcurrentSynchronizationContext.
```

Comment:
Can you explain more about what you're testing for? For example:

```cs
// Verify that the previous lock holder is allowed to continue execution when we
// as the new lock holder are blocking our own thread while it's in the NonConcurrentSynchronizationContext.
```

sp: synchorizely -> synchronously

Reply:
fixed.